packages/salesforcedx-vscode-core/src/commands/forceApexLogGet.ts

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright (c) 2017, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {
+  CliCommandExecutor,
+  Command,
+  CommandOutput,
+  SfdxCommandBuilder
+} from '@salesforce/salesforcedx-utils-vscode/out/src/cli';
+import {
+  CancelResponse,
+  ContinueResponse,
+  ParametersGatherer
+} from '@salesforce/salesforcedx-utils-vscode/out/src/types';
+import * as fs from 'fs';
+import * as path from 'path';
+import { Observable } from 'rxjs/Observable';
+import { mkdir } from 'shelljs';
+import * as vscode from 'vscode';
+import { nls } from '../messages';
+import { notificationService } from '../notifications';
+import { CancellableStatusBar, taskViewService } from '../statuses';
+import {
+  SfdxCommandlet,
+  SfdxCommandletExecutor,
+  SfdxWorkspaceChecker
+} from './commands';
+
+class ForceApexLogGetExecutor extends SfdxCommandletExecutor<
+  ApexDebugLogIdStartTime
+> {
+  public build(data: ApexDebugLogIdStartTime): Command {
+    return new SfdxCommandBuilder()
+      .withDescription(nls.localize('force_apex_log_get_text'))
+      .withArg('force:apex:log:get')
+      .withFlag('--logid', data.id)
+      .withArg('--json')
+      .build();
+  }
+
+  public async execute(
+    response: ContinueResponse<ApexDebugLogIdStartTime>
+  ): Promise<void> {
+    const cancellationTokenSource = new vscode.CancellationTokenSource();
+    const cancellationToken = cancellationTokenSource.token;
+    const execution = new CliCommandExecutor(this.build(response.data), {
+      cwd: vscode.workspace.rootPath
+    }).execute(cancellationToken);
+    this.attachExecution(execution, cancellationTokenSource, cancellationToken);
+    const resultPromise = new CommandOutput().getCmdResult(execution);
+    const result = await resultPromise;
+    const resultJson = JSON.parse(result);
+    if (resultJson.status === 0) {
+      const logDir = path.join(
+        vscode.workspace.workspaceFolders![0].uri.fsPath,
+        '.sfdx',
+        'tools',
+        'debug',
+        'logs'
+      );
+      if (!fs.existsSync(logDir)) {
+        mkdir('-p', logDir);
+      }
+      const date = new Date(response.data.startTime)
+        .toISOString()
+        .replace('T', ' ')
+        .replace('.000', '');
+      fs.writeFile(
```

Comment:
This will create a log file, but doesn't open the file automatically. I imagine it would be convenient to open the file for them. You already have the filepath and VS Code has APIs to open a text document.